code/game/machinery/doors/door.dm

```diff
@@ -272,6 +314,36 @@
 		emagged = TRUE
 		return TRUE
 
+/obj/machinery/door/cmag_act(mob/user)
+	if(density)
+		flick("door_spark", src)
+		sleep(6) //The cmag doesn't automatically open doors. It inverts access, not provides it!
+		cmagged = TRUE
+		return TRUE
+
+//Proc for inverting access on cmagged doors."canopen" should always return the OPPOSITE of the normal result.
+/obj/machinery/door/proc/cmag_switch(canopen, dooruser)
+	if(canopen && !locked)
+		if(ishuman(dooruser))
+			var/mob/living/carbon/human/H = dooruser
+			var/obj/item/card/id/idcard = H.get_idcard()
+			if(!idcard) //Humans can't game inverted access by taking their ID off.
+				if(density)
+					do_animate("deny")
+					if(hasPower())
+						to_chat(H, "<span class='warning'>The airlock speaker chuckles: 'What's wrong, pal? Lost your ID? Nyuk nyuk nyuk!'</span>")
+						playsound(loc, 'sound/machines/honkbot_evil_laugh.ogg', 25, TRUE, ignore_walls = FALSE)
```

Comment:
can you please test to see if this can be spammed?

Reply:
I've added a one second cooldown that's handled in a new proc.